### PR TITLE
fix(protocol): correct the wrong router address for mainnet

### PR DESCRIPTION
**Pull request by dantaik**

### CODE REVIEW

File: packages/protocol/contracts/layer1/mainnet/addrcache/RollupAddressCache.sol [hunk 1]
```diff
@@ -52,7 +53,7 @@ contract RollupAddressCache is AddressCache {
             return (true, address(0));
         }
         if (_name == LibStrings.B_CHAIN_WATCHDOG) {
-            return (true, address(0));
+            return (true, 0xE3D777143Ea25A6E031d1e921F396750885f43aC);
```
Comment (dantaik) [inline packages/protocol/contracts/layer1/mainnet/addrcache/RollupAddressCache.sol:56]: Please double check this.

File: packages/protocol/contracts/layer1/mainnet/rollup/MainnetTierRouter.sol [hunk 2]
```diff
@@ -2,13 +2,14 @@
...
 /// @title MainnetTierRouter
 /// @dev Labeled in AddressResolver as "tier_router"
 /// @custom:security-contact security@taiko.xyz
-contract MainnetTierRouter is ITierRouter {
+contract MainnetTierRouter is ITierRouter, TierProviderV2 {
     /// @inheritdoc ITierRouter
-    function getProvider(uint256) external pure returns (address) {
-        return 0x4cffe56C947E26D07C14020499776DB3e9AE3a23; // TierProviderV2
+    function getProvider(uint256) external view returns (address) {
+        return address(this);
```
Comment (dantaik) [inline packages/protocol/contracts/layer1/mainnet/rollup/MainnetTierRouter.sol:13]: lets deploy the router and the provider as one for convinience.

File: packages/protocol/deployments/mainnet-contract-logs-L1.md [hunk 3]
```diff
@@ -228,13 +228,9 @@
...
 
-#### tier_provider
+#### tier_router
 
-- impl: `0x3a1A900680BaADb889202faf12915F7E47B71ddd`
-- logs:
-  - deployed on May 1, 2024 @commit`56dddf2b6`
-  - deployed on May 15, 2024 @commit`cd5144255`
-  - deployed on Jul 2, 2024 @commit`d2b00ce`
+- impl: `TBD`
```
Comment (dantaik) [inline packages/protocol/deployments/mainnet-contract-logs-L1.md:233]: please update this address

File: packages/protocol/contracts/layer1/mainnet/addrcache/RollupAddressCache.sol [hunk 4]
```diff
@@ -34,7 +34,8 @@ contract RollupAddressCache is AddressCache {
             return (true, 0x06a9Ab27c7e2255df1815E6CC0168d7755Feb19a);
         }
         if (_name == LibStrings.B_TIER_ROUTER) {
-            return (true, 0x6E997f1F22C40ba37F633B08f3b07E10Ed43155a);
```
[the remark this answers is not in the document]
Comment (davidtaikocha) [reply packages/protocol/contracts/layer1/mainnet/addrcache/RollupAddressCache.sol:37]: changed to: https://etherscan.io/address/0x2cd242bfd58e6d1810323d07b065413095c7afd0

File: packages/protocol/contracts/layer1/mainnet/addrcache/RollupAddressCache.sol (hunk 1 above)
Comment (davidtaikocha) [reply packages/protocol/contracts/layer1/mainnet/addrcache/RollupAddressCache.sol:55]: That's correct.

**Review by davidtaikocha — APPROVED** [review]

**Review by YoGhurt111 — APPROVED** [review]

**Pull request merged**